### ISSUE TITLE
Minikube fails to start on Hyper-V

### ISSUE THREAD

**Issue by prdoyle** [kubernetes/minikube#1793]
**Minikube version**: v0.21.0

**Environment**:
- **OS**: Windows 10
- **VM Driver**: `hyperv`
- **ISO version**: minikube-v0.23.0.iso
- **Install tools**: ?

**What happened**:
I did once successfully run `minikube start --vm-driver=hyperv`.  I shut it down, restarted, and even once had to safe-boot and delete `.minikube` but to no avail: it's never worked again.

Now I get this:

```
Microsoft Windows [Version 10.0.15063]
(c) 2017 Microsoft Corporation. All rights reserved.

C:\WINDOWS\system32>minikube start --vm-driver=hyperv
Starting local Kubernetes v1.7.0 cluster...
Starting VM...
Downloading Minikube ISO
 97.80 MB / 97.80 MB [==============================================] 100.00% 0s
E0805 22:14:37.098639    9904 start.go:129] Error starting host: Error creating host: Error executing step: Creating VM.
: exit status 1.

 Retrying.
E0805 22:14:37.102237    9904 start.go:135] Error starting host:  Error creating host: Error executing step: Creating VM.
: exit status 1
================================================================================
An error has occurred. Would you like to opt in to sending anonymized crash
information to minikube to help prevent future errors?
To opt out of these messages, run the command:
        minikube config set WantReportErrorPrompt false
================================================================================
Please enter your response [Y/n]:
```

**What you expected to happen**:
I expected the cluster to come up.

**How to reproduce it** (as minimally and precisely as possible):
On Windows 10, follow the install instructions and run Minikube.  It should work the first time.  After that, shut it down.  If you try to run it again, and you're unlucky like me, it might not work ever again.

**Anything else do we need to know**:
I'm running Babun and once tried `minikube start` from a bash prompt.  Not sure whether that's relevant.

I'm also running Docker for Windows, but I don't think that's the problem because it did work once even with Docker for Windows running.

**Comment by whereisaaron** [comment]
Hi @prdoyle you may have missed a step for Hyper-V:

https://github.com/kubernetes/minikube/blob/master/docs/drivers.md#hyperv-driver

You need to create an 'External address' 'Virtual Switch' in Hyper-V:

https://docs.docker.com/machine/drivers/hyper-v/

and then name that when you first start your minikube:

`minikube start --vm-driver hyperv ----hyperv-virtual-switch "Primary Virtual Switch"`

I suggest you create the switch, and then delete and start minikube with the switch setting:

```
minikube delete
minikube start --vm-driver hyperv ----hyperv-virtual-switch "Primary Virtual Switch"
```

Running Docker for Windows at the same time is no problem. Using minikube.exe from the bash/WSL prompt is no problem.

**Comment by prdoyle** [comment]
Excellent, thanks so much for your help.  I'm trying it right now.

Question: is it normal that `minikube delete` does this?

```
{ ~ }  » minikube delete                                                                                                                                  Deleting local Kubernetes cluster...
Errors occurred deleting machine:  exit status 1
remove C:\Users\pdoyle\.babun\cygwin\home\pdoyle\.minikube\machines\minikube\disk.vhd: The process cannot access the file because it is being used by another process.
```

I find that I frequently need to reboot into safe mode so I can delete the `.minikube` directory, which is kind of a pain.

**Comment by prdoyle** [comment]
Still not working.  I rebooted in safe mode and deleted `.minikube`, then tried again and got this:

```
{ ~ }  » minikube start --vm-driver hyperv --hyperv-virtual-switch "external-switch"                                                                        ~
Starting local Kubernetes v1.7.0 cluster...
Starting VM...
E0807 13:15:17.713639   11752 start.go:129] Error starting host: Error loading existing host. Please try running [minikube delete], then run [minikube start] again.: Error loading host from store: open C:\Users\pdoyle\.babun\cygwin\home\pdoyle\.minikube\machines\minikube\config.json: The system cannot find the file specified..

 Retrying.
E0807 13:15:17.715649   11752 start.go:135] Error starting host:  Error loading existing host. Please try running [minikube delete], then run [minikube start] again.: Error loading host from store: open C:\Users\pdoyle\.babun\cygwin\home\pdoyle\.minikube\machines\minikube\config.json: The system cannot find the file specified.
```

**Comment by whereisaaron** [comment]
I am not sure deleting `.minikube` is a good idea, the Hype-V VM will still be there orphaned. This probably causes you subsequent problems. Nor do you ever need to reboot in safe mode. I suggest you stop the minikube before you try and delete it. And if you orphaned a VM, go into Hyper-V console and stop and delete it manually.

```
minikube stop
minikube delete
minikube start --vm-driver hyperv --hyperv-virtual-switch "external-switch"
```

**Comment by mmisztal1980** [comment]
I'm having a similar issue:

Initially, after a restart I've done:

```
minikube stop
minikube delete
```
**.minikube deleted**

```
minikube start --vm-driver hyperv --hyperv-virtual-switch External --alsologtostderr
```

And I've got:

```
minikube start --vm-driver hyperv --hyperv-virtual-switch External --alsologtostderr
W0809 13:09:09.422088   13224 root.go:138] Error reading config file at C:\Users\mmisztal\.minikube\config\config.json: open C:\Users\mmisztal\.minikube\config\config.json: The system cannot find the path specified.
I0809 13:09:09.434088   13224 notify.go:112] Checking for updates...
Starting local Kubernetes v1.7.0 cluster...
Starting VM...
Downloading Minikube ISO
 97.80 MB / 97.80 MB [==============================================] 100.00% 0s
```

Seems that It's stuck for some reason after downloading the ISO.
If the ISO is already present on disk, it gets stuck faster.

Can someone advise?

**Comment by prdoyle** [comment]
@whereisaaron - Thanks for the newbie help on Hyper-V!  I cleaned up the orphaned VMs, deleted `.minikube` again, and this time `minikube start` worked!

( I went against your advice and deleted `.minikube` because `minikube stop` and  `minikube delete` gave errors for me.)

**Comment by r2d4** [comment]
Closing, looks like @prdoyle resolved the issue.

**Comment by prdoyle** [comment]
@r2d4 - do you think I should open a new Issue regarding the fact that it seems necessary sometimes to delete the `.minikube` directory and start fresh?

**Comment by r2d4** [comment]
Sure - feel free to.  There are still a few cases where a minikube start might error and put the machine VM in an irreconcilable place with the config.  We should try to fix those cases.

**Comment by amantur** [comment]
well, I ran `stop` and `delete` commands and got error as below:

`C:\>minikube stop
Stopping local Kubernetes cluster...
Error stopping machine:  Error loading host: minikube: Error loading host from store: Host does not exist: "minikube"

C:\>minikube delete
Deleting local Kubernetes cluster...
Errors occurred deleting machine:  Error deleting host: minikube: Error loading host from store: Host does not exist: "minikube"`
 
But running again with the switch name without quotes worked:

`minikube start --vm-driver hyperv --hyperv-virtual-switch minikube-ext`

**Comment by yuanl2** [comment]
I meet the issue on windows 10.
I create  an 'External address' 'Virtual Switch' in Hyper-V, but looks not working.
There is still error. So guys, do you have any suggestions? Thank you.


Starting local Kubernetes v1.9.0 cluster...
Starting VM...
Downloading Minikube ISO
 142.22 MB / 142.22 MB  100.00% 0ss0s
E0205 15:23:15.979614    8628 start.go:159] Error starting host: Error creating host: Error executing step: Creating VM.
: exit status 1.

 Retrying.
E0205 15:23:15.983627    8628 start.go:165] Error starting host:  Error creating host: Error executing step: Creating VM.
: exit status 1
================================================================================
An error has occurred. Would you like to opt in to sending anonymized crash
information to minikube to help prevent future errors?
To opt out of these messages, run the command:
        minikube config set WantReportErrorPrompt false
================================================================================

**Comment by yuanl2** [comment]
@gbraad , thank you for your suggestion. It can work.

**Comment by chrisxaustin** [comment]
This happened for me when I did the following:

In a Cmder admin instance:
```minikube start  --vm-driver hyperv --hyperv-virtual-switch "minikube"```

In a Cmder bash instance:
```minikube docker-env```
Which failed, then I tried `minikube delete` from the bash instance, which gave the above error.
I then tried `minikube delete` from the Cmder admin instance, which gave that error.
Deleting the Hyper-V vm didn't clear it up, but deleting ~/.minikube did.

**Comment by binititpro** [comment]
For windows-7, please set environment variable as MINIKUBE_HOME=c:\users\***\.minikube.
This is work for me.

**Comment by umartunes** [comment]
> Hi @prdoyle you may have missed a step for Hyper-V:
> 
> https://github.com/kubernetes/minikube/blob/master/docs/drivers.md#hyperv-driver
> 
> You need to create an 'External address' 'Virtual Switch' in Hyper-V:
> 
> https://docs.docker.com/machine/drivers/hyper-v/
> 
> and then name that when you first start your minikube:
> 
> `minikube start --vm-driver hyperv ----hyperv-virtual-switch "Primary Virtual Switch"`
> 
> I suggest you create the switch, and then delete and start minikube with the switch setting:
> 
> ```
> minikube delete
> minikube start --vm-driver hyperv ----hyperv-virtual-switch "Primary Virtual Switch"
> ```
> Running Docker for Windows at the same time is no problem. Using minikube.exe from the bash/WSL prompt is no problem.

It worked for me. Thanks!

**Comment by mmoser18** [comment]
> 
> @whereisaaron - Thanks for the newbie help on Hyper-V! I cleaned up the orphaned VMs, deleted `.minikube` again, and this time `minikube start` worked!
> 
> ( I went against your advice and deleted `.minikube` because `minikube stop` and `minikube delete` gave errors for me.)

Same BS here! minikube wouldn't move forward nor backward. No stop, no delete, no start, nothing worked! ALL commands just gave the dreaded "? Errors occurred <whatever command I tried...>:  exit status 1
Only after deleting .minikube I finally got going again!

Thanks for the hint!

**Comment by afbjorklund** [comment]
Disregard mention, wrong project.

**Comment by anshuld27** [comment]
> minikube stop
> minikube delete
> minikube start --vm-driver hyperv --hyperv-virtual-switch "external-switch"

This is working for me.
